src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -610,46 +610,43 @@ private bool TryDecodeMime(StringSegment input, [NotNullWhen(true)] out string?
         private static string Encode5987(StringSegment input)
         {
             var builder = new StringBuilder("UTF-8\'\'");
-            var unicodeCharsCount = input.Value.Count(c => c > 0x7F);
-            var bufferSize = unicodeCharsCount > 0
-                ? Encoding.UTF8.GetMaxByteCount(unicodeCharsCount)
-                : 0;
-            Span<byte> buffer = bufferSize > 0
-                ? bufferSize <= 256
-                    ? stackalloc byte[bufferSize]
-                    : new byte[bufferSize]
-                : null;
+            ReadOnlySpan<byte> inputBytes = Encoding.UTF8.GetBytes(input.Value);
```

Comment:
This allocates a byte array, that then is assigned to the span.

How about:
```c#
[SkipLocalsInit]
private static string Encode5987(string input)
{
    var maxInputBytes = Encoding.UTF8.GetMaxByteCount(input.Length);
    byte[]? bufferFromPool = null;
    Span<byte> inputBytes = maxInputBytes <= 256    // or use any other already defined const
        ? stackalloc byte[256]
        : bufferFromPool = ArrayPool<byte>.Shared.Rent(maxInputBytes);

    var bytesWritten = Encoding.UTF8.GetBytes(input, inputBytes);
    inputBytes = inputBytes[..bytesWritten];

    var totalBytesConsumed = 0;
    while (totalBytesConsumed < inputBytes.Length)
    // ...
    // code as is
    // ...

    if (bufferFromPool is not null)
    {
        ArrayPool<byte>.Shared.Return(bufferFromPool);
    }

    return builder.ToString();
}
```
More ceremony, but less allocations...

Reply:
Thanks, @gfoidl! This should improve our benchmarks even more, applying update.